### PR TITLE
Fix docker-machine.rb

### PR DESCRIPTION
**Pull request by novikovSU**
Upgrade type of container to zip-archive and replace one binary to list of binaries.

In current version of Docker Machine 0.5.0 they split functionality to a main command and a bunch of drivers (separate binaries). Also they put it all in the zip-archive.

### CODE REVIEW

File: Casks/docker-machine.rb [hunk 1]
```diff
@@ -9,11 +9,42 @@
   homepage 'https://docs.docker.com/machine'
   license :apache
 
-  container :type => :naked
-  binary 'docker-machine_darwin-amd64', :target => 'docker-machine'
+  container :type => :zip
```
Comment (vitorgalvao) [inline Casks/docker-machine.rb]: Are you sure this is needed?

Comment (kongslund) [inline Casks/docker-machine.rb]: Currently the installation fails so _something_ different is needed.

````
$ brew cask install docker-machine
==> Satisfying dependencies
==> Installing Formula dependencies from Homebrew
docker ... already installed
complete
==> Downloading https://github.com/docker/machine/releases/download/v0.5.0/docker-machine_darwin-amd64.zip
Already downloaded: /Library/Caches/Homebrew/docker-machine-0.5.0.zip
Error: It seems the symlink source is not there: '/opt/homebrew-cask/Caskroom/docker-machine/0.5.0/docker-machine_darwin-amd64'
````

Comment (kongslund) [inline Casks/docker-machine.rb]: ...but the proposed fix does not not work for me:
````
$ brew cask install docker-machine
==> Satisfying dependencies
==> Installing Formula dependencies from Homebrew
docker ... already installed
complete
==> Downloading https://github.com/docker/machine/releases/download/v0.5.0/docker-machine_darwin-amd64.zip
Already downloaded: /Library/Caches/Homebrew/docker-machine-0.5.0.zip
==> chmod: /opt/homebrew-cask/Caskroom/docker-machine/0.5.0/docker-machine_darwin-amd64: No such file or directory
Error: Command failed to execute!

==> Failed command:
["/usr/bin/sudo", "-E", "--", "/bin/chmod", "-R", "--", "0755", "#<Pathname:/opt/homebrew-cask/Caskroom/docker-machine/0.5.0/docker-machine_darwin-amd64>"]

==> Output of failed command:


==> Exit status of failed command:
#<Process::Status: pid 13783 exit 1>
````

This is my patched version:
````
$ brew cask cat docker-machine
cask :v1_1 => 'docker-machine' do
  version '0.5.0'
  sha256 '75f18940c734425b3635c9485da148910dee6612cc677549563ba7662e3d36f6'

  # github.com is the official download host per the vendor homepage
  url "https://github.com/docker/machine/releases/download/v#{version}/docker-machine_darwin-amd64.zip"
  appcast 'https://github.com/docker/machine/releases.atom'
  name 'Docker Machine'
  homepage 'https://docs.docker.com/machine'
  license :apache

  #container :type => :naked
  #binary 'docker-machine_darwin-amd64', :target => 'docker-machine'
  container :type => :zip

  postflight do
    set_permissions "#{staged_path}/docker-machine_darwin-amd64", '0755'
  end

  depends_on :formula => 'docker'
  depends_on :arch => :x86_64
end
````

File: Casks/docker-machine.rb [hunk 2]
```diff
@@ -9,11 +9,42 @@
...
+  binary 'docker-machine-driver-none'
+  binary 'docker-machine-driver-openstack'
+  binary 'docker-machine-driver-rackspace'
+  binary 'docker-machine-driver-softlayer'
+  binary 'docker-machine-driver-virtualbox'
+  binary 'docker-machine-driver-vmwarefusion'
+  binary 'docker-machine-driver-vmwarevcloudair'
+  binary 'docker-machine-driver-vmwarevsphere'
 
   postflight do
```
Comment (jawshooah) [inline Casks/docker-machine.rb]: This block is no longer necessary, as all these files are already executable.

File: Casks/docker-machine.rb (hunk 1 above)
Comment (jawshooah) [inline Casks/docker-machine.rb]: We don't need a `container` stanza at all. This works:

```ruby
cask :v1_1 => 'docker-machine' do
  version '0.5.0'
  sha256 '75f18940c734425b3635c9485da148910dee6612cc677549563ba7662e3d36f6'

  # github.com is the official download host per the vendor homepage
  url "https://github.com/docker/machine/releases/download/v#{version}/docker-machine_darwin-amd64.zip"
  appcast 'https://github.com/docker/machine/releases.atom'
  name 'Docker Machine'
  homepage 'https://docs.docker.com/machine'
  license :apache

  binary 'docker-machine'
  binary 'docker-machine-driver-amazonec2'
  binary 'docker-machine-driver-azure'
  binary 'docker-machine-driver-digitalocean'
  binary 'docker-machine-driver-exoscale'
  binary 'docker-machine-driver-generic'
  binary 'docker-machine-driver-google'
  binary 'docker-machine-driver-hyperv'
  binary 'docker-machine-driver-none'
  binary 'docker-machine-driver-openstack'
  binary 'docker-machine-driver-rackspace'
  binary 'docker-machine-driver-softlayer'
  binary 'docker-machine-driver-virtualbox'
  binary 'docker-machine-driver-vmwarefusion'
  binary 'docker-machine-driver-vmwarevcloudair'
  binary 'docker-machine-driver-vmwarevsphere'

  depends_on :formula => 'docker'
  depends_on :arch => :x86_64
end
```

**Comment by jawshooah** [comment]
Thank you for the contribution. It needed some fixes, so they were made in a1b82a20d1c1727bee340ce5db09c56a4dac2415. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

File: Casks/docker-machine.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/docker-machine.rb]: > This is my patched version:
> (…)
> #container :type => :naked

That’s what I meant. Like @jawshooah pointed out, `container :type` isn’t needed at all. Even you were removing it, so why add it back with a different Value?

Comment (kongslund) [inline Casks/docker-machine.rb]: Nevermind my comment about the fix not working. My manual patching was sadly incomplete.

**Comment by novikovSU** [comment]
Hi, guys!
I'm new to Homebrew Cask so sorry if my proposed changes is not so good. It's works for me. :) I'm glad what you'd fixed the problem.

**Pull request closed without merging**